Feature/nkou html

beforeisland opened the pull request:
N高のページ模写課題のHTML部分のコーディングが完了しました。

beforeisland committed:
add: initial commit

beforeisland committed:
add: head設定

head設定を行いました。

beforeisland committed:
add: ヘッダー部分のhtml実装

ヘッダー部分の固定メニュー、ナビリンク部分を実装しました

beforeisland committed:
add: メイン記事部分メイン画像・パンくずナビhtml実装

メイン記事部分の画像及びパンくずナビを実装しました。

beforeisland committed:
fix: ナビ部分セクション分け変更

ナビ部分のセクション分けについて、ulタグの親要素としてnavタグを追加修正。

beforeisland committed:
add: メイン記事html実装

メイン記事・教育方針・N高の魅力のブロックのhtmlを実装しました。

beforeisland committed:
add: ベーシックプログラム部分html実装

メイン記事のベーシックプログラム部分のhtmlを実装しました。

beforeisland committed:
add: アドバンスドプログラム部分html実装

メイン記事・アドバンスドプログラム部分のhtmlを実装しました。

beforeisland committed:
add: itツール部分のhtml実装

メイン記事のitツール部分のhtmlを実装しました。

beforeisland committed:
add: 学校行事部分のhtml実装

メイン記事の学校行事部分のhtmlを実装しました。

beforeisland committed:
add: N高起業部の部分のhtml実装

メイン記事のN高起業部の部分のhtmlを実装しました。

beforeisland committed:
add: 部活動部分のhtml実装

メイン記事の部活動部分のhtmlを実装しました。

beforeisland committed:
add: 入学コース部分のhtml実装

メイン記事の入学コース部分のhtmlを実装しました。

beforeisland committed:
add: 提携スクール部分のhtml実装

メイン記事の提携スクール部分のhtmlを実装しました。

beforeisland committed:
add: 補足記事・N高バナーリンクhtml実装

メイン記事の最後にあたるN高バナーリンクのhtmlを実装しました。

beforeisland committed:
add: メッセージ記事部分html実装

メイン記事のメッセージ記事部分のhtmlを実装しました。

beforeisland committed:
add: N高についての詳細リンク記事html実装

補足記事部分である、詳細リンク記事のhtmlを実装しました。

beforeisland committed:
add: 補足記事部分のニュース、その他の教育記事部分html実装

補足記事部分のニュース・ブログ・N中等部・Nコードラボのリンク記事部分を実装。

beforeisland committed:
add: フッター部分ローカルナビhtml実装

フッター部分のローカルナビ、SNSリンク他html実装

beforeisland committed:
add: コピーライト部分のhtml実装

フッター部分のコピーライト部分のhtmlを実装しました。

beforeisland committed:
fix: htmlチェッカー使用後の修正

一部画像のalt属性未実装の修正、セクショニングの修正、リストのタグ構成エラー修正

choimake commented on index.html:
```diff
@@ -0,0 +1,795 @@
+        content="N高等学校（通信制高校）は、IT×グローバル社会を生き抜く「創造力」身につけ、世界で活躍する人を育成します。N高の教育方針とネットの高校だからこそ学べる魅力を紹介します。">
+    <title>N高について | N高等学校(通信制高校 広域・単位制)</title>
+    <link rel="stylesheet" href="common/css/normalize.css">
+    <link rel="stylesheet" href="common/css/style.css">
```
存在しないcssを読み込む処理は消しておいた方が良いです。
後々実装する際にどうせ書くからということだとは思うんですが、ブラウザがエラーを吐く挙動のままレビューに出すのはNGとします。
仮に何かしらの不具合が発生しそれを解決するとなったときに、エラーの頭数が多いと、エラーの原因の切り分けが難しくなってしまうので。
エラーの数がゼロの状態を保つようにしましょう。

beforeisland replied:
本件、承知いたしました。エラーの遠因となる可能性は排除しておきます。

fix: スタイルシート読み込みリンク削除	ef4a83e	beforeisland <beforeisland@gmail.com>	2020/07/29 20:31

choimake commented on index.html:
```diff
@@ -0,0 +1,795 @@
+            <span>About School</span>
+        </div>
+        <img class="main-visual" src="common/images/about_mainimage01.jpg" alt="メインイメージ">
+        <nav class="topic-path">
+            <a href="#">N高等学校 ｜ 通信制高校(広域)・単位制</a> &gt; N高について
+        </nav>
```
ここはパンくずリストに当たる部分なので、パンくずリストとして機能するようなhtmlで実装してください。

beforeisland replied:
パンくずリストの正しい実装方法の知識がなかったため、下記を参照しリストを使ってURLを階層構造にして対応しました。
https://webliker.info/94182/

fix: パンくずリスト実装方法変更	c40d762	beforeisland <beforeisland@gmail.com>	2020/07/29 20:44

choimake commented on index.html:
```diff
@@ -0,0 +1,795 @@
+        </div>
+        <div class="local-navi-container">
+            <nav class="local-navi-list">
+                <p>
+                    <a href="#">
+                        <span>アドバンスト プログラム</span><br>
+                        <span>Advanced Program</span>
+                    </a>
+                </p>
```
pは段落という意味ですが、ここは本当に段落なのでしょうか？
（同様に似たような実装をしている箇所があるので合わせて確認してください）

beforeisland replied:
ご指摘頂いたとおり段落ではありません。CSSでの装飾のことを考慮して実装していました。リンクナビのリストに組み込む方式に変更し、CSS実装時にCSS側から要素の先頭指定して装飾を実装します。

fix: ローカルナビ部分の太字リンク実装方法変更	35da616	beforeisland <beforeisland@gmail.com>	2020/07/29 21:01

choimake replied:
参考までに、「htmlを実装する段階では装飾のためのタグは実装しなくて良い」と考えています。
理由ですが、htmlでは文章構造を実装することが最も優先される大事な目的で、装飾のためのタグの実装は装飾の時にやれば良いからです。

文章構造を実装するときに装飾の実装を一緒にやってしまうと、文章構造の実装がなおざりになったりすることがあります。
まとめて実装可能ならやっても良いっちゃあ良いですが、誰かにレビューしてもらう環境があるときや、装飾のことを考えて文章構造がなおざりになってしまう場合は、文章構造のhtmlのみ実装するようなやり方をしたほうが無難かなと考えます。

このあたりの正誤は人それぞれだし、環境にもよるので、あくまで参考程度にみておいてくれれば幸いです。

choimake commented on index.html:
```diff
@@ -0,0 +1,795 @@
+    </footer>
+
+
+</body>
```
ファイルの末尾には改行を入れましょう。

Githubでファイル末尾に改行がないと、赤いマークが出ます。

何故かと言いますと、
「テキストファイルは本来（POSIX的には）必ず最後は改行コードで終わっていなければならない」というルールがあるからです。

それが守られていないときにGithubでは赤いマークという形で警告がでます。

以下、参考URLです。
https://www.cqpub.co.jp/interface/sample/200902/if02_038.pdf

標準規格やIEEE、POSIXなどについて一度確認してみてください。

beforeisland replied:
ソフトウェアの標準規格は知識がありませんでした。参考資料ありがとうございます。
htmlの閉じタグ忘れもあったため、バリデーションエラーと併せて修正しました。

fix: バリデーションエラー修正	af0b6ea	beforeisland <beforeisland@gmail.com>	2020/07/29 21:05

choimake reviewed (commented):
NRIの課題のときと比べて、色々と洗練され、良くなっています。

ただいくつか気になる点があったので、以下にコメントしました。
確認お願いします。

beforeisland committed:
fix: スタイルシート読み込みリンク削除

現時点ではCSSを書いておらず、万が一エラーが起きた場合に原因の遠因となる可能性も考えられるため削除

beforeisland committed:
fix: パンくずリスト実装方法変更

パンくずリストの実装方法を、リストを使ってURLを階層構造にする方法に変更。

beforeisland committed:
fix: ローカルナビ部分の太字リンク実装方法変更

ローカルナビの太字リンク部分について、pタグを使用してマークアップしていたが、段落ではないため実装方法を変更。リストに組み込み、装飾はCSSで指定を切り分ける予定。

beforeisland committed:
fix: バリデーションエラー修正

aタグ閉じタグ間違い修正、html閉じタグ未実装修正、ファイルエンドに空欄実装

choimake reviewed (approved):
修正ありがとうございました。
LGTMです。次に進んでください。
